### PR TITLE
Container proxy

### PR DESCRIPTION
**Pull request by bqqbarbhg**
Add proxy for routing requests from subdomain to user app containers. Added also initial database schema for containers table. The constraint was added to a previous migration as SQLite can't handle adding constraints later to tables, we need to figure out what to do about this.

The default settings for the Go proxy work with the development Node server, ie. compatible with `cd server && npm start`. The proxy doesn't support Postgres yet, but it should literally be one line change in the imports of `resolver.go`.

The proxy uses VGo modules so you need at least Go 1.11 to build this. With a compatible Go version building and running should work as simply as:
```bash
cd proxy
go build
./proxy
```

Note that if you are using Windows you may have issues with building the sqlite3 dependency as it's a CGo package that requires `gcc` to be in path. Moreover you may need 64-bit `gcc` (at least I did). I got it working with Mingw64: https://sourceforge.net/projects/mingw-w64/

### CODE REVIEW

**Comment by bqqbarbhg** [comment]
Oh damn tests broke due to adding the `FOREIGN KEY` constraint to the threads table, will fix tomorrow!

File: proxy/resolver/resolver.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+// -- Database resolver
+
+var db *sql.DB
+
+// Pending requests to the database
+var databaseRequests = make(chan resolveRequest)
+
+// Query the database and create a container instance
+func fetchContainerFromDatabase(id string) (*Container, error) {
+	row := db.QueryRow("SELECT subdomain, url FROM containers WHERE subdomain = ?", id)
```
Comment (tobbe-j) [inline proxy/resolver/resolver.go]: Voi tulla ongelmia Postgres:in kanssa, nodessa postgres käytti $1,$2... '?','?' sijaan

Comment (bqqbarbhg) [reply proxy/resolver/resolver.go]: Will test when I can merge Postgres here

File: server/src/database/migrations/002-containers.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+-- Text encoding used: UTF-8
+--
+BEGIN TRANSACTION;
+
+-- Table: containers
```
Comment (tobbe-j) [inline server/src/database/migrations/002-containers.sql]: Lisää myös database docsiin

**Review by tobbe-j — CHANGES_REQUESTED** [review]
Docs

File: server/src/database/migrations/002-containers.sql (hunk 2 above)
Comment (bqqbarbhg) [reply server/src/database/migrations/002-containers.sql]: Done

**Review by tobbe-j — DISMISSED** [review]

**Comment by bqqbarbhg** [comment]
Now this PR should be in a working condition.

Changes:
- Added support for using the proxy in a development server
- Added `--startProxy` argument to server for starting the proxy executable as a child process
- Added support for using the proxy in the development Docker build
- Added Travis CI tests for the proxy

**Comment by bqqbarbhg** [comment]
Also includes the following misc fixes:
- Only initialize the feedback tool once even if it's imported multiple times via `<script>`
- Retry Postgres migrations a few times with timeout in case the DB server is still booting

**Review by tobbe-j — APPROVED** [review]
Looks good

**Comment by HenrikHelenius** [comment]
[(25/89) [task] [dynamic docker] Make dynamic docker Samuli](https://trello.com/c/LN926OBv/129-25-89-task-dynamic-docker-make-dynamic-docker-samuli)

**Pull request merged**